_posts/2018-11-20-docker-setup.md

```diff
@@ -0,0 +1,320 @@
+    ```
+
+2. Build Docker containers (This will take a while).
+
+    ```bash
+    → docker-compose build
+    ```
+
+3.  Run Docker containers (This will take a while as well).
+
+    ```bash
+    → docker-compose up
+    ```
+
+    If everything goes well, you should see all the containers running.
+
+    ```bash
+    → docker ps
+    ```
+
+    ```bash
+    CONTAINER ID        IMAGE                      COMMAND                  CREATED             STATUS              PORTS                    NAMES
+    b82c59b64d2e        docker_sidekiq   "bundle exec sidekiq  "    57 seconds ago      Up 56 seconds                                sidekiq
+    3ed5cf88d125        docker_worker    "bundle exec rails s  "   57 seconds ago      Up 56 seconds       0.0.0.0:3002->3000/tcp   worker
+    7e92bb3df879        docker_manager   "bundle exec rails s  "   57 seconds ago      Up 56 seconds       0.0.0.0:3001->3000/tcp   manager
+    ae32635f4802        docker_api       "bundle exec rails s  "   58 seconds ago      Up 57 seconds       0.0.0.0:3000->3000/tcp   api
+    dad90c3e0039        redis:2.8         "docker-entrypoint.sh"   58 seconds ago      Up 57 seconds       0.0.0.0:6379->6379/tcp   redis
+    a4e536734a7b        docker_solr      "bundle exec rake su.."    59 seconds ago      Up 57 seconds       0.0.0.0:8983->8983/tcp   solr
+    47fba6738e23        mongo:3.6.8      "/entrypoint.sh mongo"   17 hours ago        Up 57 seconds       27017/tcp                supplejackdocker_mongodb_1
+    ```
+
+4. If you exit the docker process for any reason, make sure you run `docker-compose down` before running `docker-compose up` again.  This will ensure the PID'S are correctly reset.
+
+### Seeding Data
+
+The Supplejack components are connected by API keys. Before start using it, make sure to run the following commands to generate default users.
+
+Make two users in the api:
+1. a user for the harvester App (Supplejack Manager) in the **sample Api container**.
+2. a normal api user in the sample Api container
+```ruby
+$ docker-compose exec api rails c
+
+irb(main):001:0> SupplejackApi::User.create(email: 'info@email.com', password: 'password', password_confirmation: 'password', role: 'harvester', authentication_token: 'KJ64DC023FFO', name: 'harvester User', username: 'harvester User')
+
+irb(main):002:0> SupplejackApi::User.create(email: 'developer@email.com', password: 'password', password_confirmation: 'password', role: 'developer', authentication_token: '82HYSEI92N0DGN28', name: 'Mr Jones', username: 'jonesy')
+
+irb(main):003:0>exit
+```
+
+
+Make a user in the Supplejack Manager container with the following command (this is the user you will log in with:
+
+```ruby
+$ docker-compose exec manager rails c
+
+irb(main):001:0> User.create(email: 'developer@email.com', name: 'Mr Jones', password: 'password', password_confirmation: 'password', role: 'admin')
+irb(main):002:0> exit
+```
+
+This will generate the following user and keys.
```

Comment:
Is this an example of what is generated from running these commands, or is this a file of where the generated keys are stored?

Reply:
I believe it's an example of what is generated from running the commands